packages/core/src/utils.ts

```diff
@@ -683,6 +683,24 @@ export function isLowerCase(c: string): boolean {
   return c === c.toLowerCase() && c !== c.toUpperCase();
 }
 
+export function isComplexTypeCode(code: string): boolean {
```

Comment:
Here "complex type" just means "not a primitive"?

I know this came up in Slack the other day, but we may want to do a sanity check on `System.String`, which may not behave as expected here.  I believe (??) that we normalize `System.String` to `string` in the schema loader, but that might be worth double checking.

Reply:
Yeah, I added a test `expect(isComplexTypeCode(PropertyType.SystemString)).toEqual(false);` to make sure it's handled correctly. It didn't require any special handling in the function itself since `PropertyType.SystemString === 'http://hl7.org/fhirpath/System.String'` which does indeed start with a lower-case letter 🙃

Reply:
ah, right, lol